src/analyze/BuildSystem.re

```diff
@@ -130,32 +130,42 @@ let detect = (rootPath, bsconfig) => {
 let getEsyCompiledBase = (root) => {
   let env = Unix.environment()->Array.to_list;
 
-  switch(Utils.getEnvVar(~env, "cur__original_root"), Utils.getEnvVar(~env, "cur__target_dir")) {
```

Comment:
Sorry for the big diff here! I remove this first block that checks the environment for `esy` environment variables, so it impacted the indentation.

This caused problems specifically in our test environment - it would pick up the esy root/target for `reason-language-server` instead of the test case. The `esy command-env --json` strategy seems more reliable, so I just deferred to that. @anmonteiro - let me know if I missed something here!

Reply:
@bryphe checking for the env var is there to support esy named sandboxes, i.e. having `foo.json` instead of `esy.json` and starting your editor with `esy @foo code .`. I use them pretty heavily personally and this is the only way we can support them right now.

I think this needs to stay there. Can you think of an alternative?

Reply:
Thanks for the details @anmonteiro - supporting sandboxes makes sense.

I believe the issue here is limited to our tests - when we run `esy dune exec ExamplesTests`, it picks up the `cur__original_root` and `cur__target_dir` from the root `reason-language-server` esy config - this causes problems w/ the test.

A couple possible options:
- We might be able to clear out the `cur__original_root`/`cur__target_dir` environment via the `setenv` stanza in dune
- Alternatively, we could set an environment variable that we're in a test environment, and skip the sandbox that way.

Still looking into this but let me know if you have ideas. I think it's crucial to have a test covering the `esy+dune` scenario here.

Reply:
Looks like `esy` wasn't being launched in the correct spot. I think we'll also have to run `ExamplesTest.exe` _outside_ the esy sandbox, otherwise the environment gets polluted with `cur__original_root`/`cur__target_dir`

Reply:
Ah I think that makes sense. I'm glad the issue seems to be limited only to tests.